dask/utils.py

```diff
@@ -399,14 +402,21 @@ def __call__(self, arg):
             pass
         else:
             register()
-            return self(arg)  # recurse
+            return self  # recurse
```

Comment:
I just want to confirm, either case works correct? (logically to me looks fine) I even tested with `register_lazy` and I don't see an issue.
Basically, I ask this because if there is a logic difference, then maybe it's a good opportunity to add a test, thanks!